packages/tizen-remote/test/e2e/tizen-remote.e2e.spec.js

```diff
@@ -0,0 +1,364 @@
+/* eslint-disable promise/no-native */
+
+/**
+ * This test suite checks the interaction of the lib with a WebSocket server.
+ *
+ * If the following env variables are present, the test suite will use them:
+ *
+ * - `TEST_TIZEN_REMOTE_HOST`: The host of the WebSocket server.
+ * - `TEST_TIZEN_REMOTE_PORT`: The port of the WebSocket server.
+ *
+ * If the above env variables are _not_ present, a mock server will be started on a random port.
+ *
+ * This is optional:
+ *
+ * - `TEST_TIZEN_REMOTE_TOKEN`: The token to use for the WebSocket server
+ *
+ * Tokens are associated with "names", and the default name in this module is `Appium`.
+ *
+ * @module
+ */
+
+import {createSandbox} from 'sinon';
+import getPort from 'get-port';
+import {TestWSServer} from './server';
+import unexpected from 'unexpected';
+import {constants, Event, Keys, TizenRemote} from '../../lib/index';
+import d from 'debug';
+import unexpectedSinon from 'unexpected-sinon';
+import unexpectedEventEmitter from 'unexpected-eventemitter';
+import {Env} from '@humanwhocodes/env';
+
+const env = new Env();
+
+const HOST = /** @type {string} */ (env.get('TEST_TIZEN_REMOTE_HOST', '127.0.0.1'));
+const PORT = env.get('TEST_TIZEN_REMOTE_PORT');
+const TOKEN = env.get('TEST_TIZEN_REMOTE_TOKEN');
+
+const debug = d('tizen-remote:test:e2e:ws');
+
+const expect = unexpected.clone().use(unexpectedSinon).use(unexpectedEventEmitter);
+
+describe('websocket behavior', function () {
+  /** @type {TestWSServer} */
+  let server;
+
+  /** @type {number} */
+  let port;
+
+  /** @type {TizenRemote} */
+  let remote;
+
+  /** @type {sinon.SinonSandbox} */
+  let sandbox;
+
+  /** @type {import('../../lib').TizenRemoteOptions} */
+  let remoteOpts;
+
+  /** @type {string} */
+  let token;
+
+  before(async function () {
+    if (PORT) {
+      this.timeout('1m');
+      port = Number(PORT);
+      if (TOKEN) {
+        token = TOKEN;
+      } else {
+        debug('[SETUP] Getting token from %s:%d; this may be slow...', HOST, port);
+        const remote = new TizenRemote({
+          host: HOST,
+          port
+        });
+        token = await new Promise((resolve, reject) => {
```

Comment:
would be much cleaner to `await remote.getToken()`, no?

Reply:
or can't we just `await remote.connect()`?

Reply:
`remote.connect()` resolves with a `WebSocket`, not a `token`, so if we needed the actual token we'd need to call `getToken()`.  the reason for doing it here is that it's slow, and we only want to do it once.  if we allow it to happen during some other call to `remote.connect()` in the tests, then the execution time of that other call will necessarily be longer.  anyway, that's why I do it in `before()`, and that's why I increase the timeout only if we're going to go get the token.

anyway: I think I can just call `getToken()` here instead.